`tools` tests don't locate python version correctly.

I'm trying to run the statsmodels test suite to fix a reported scipy bug (https://github.com/scipy/scipy/issues/11367).

I setup a docker container to run the test suite but the container has different python versions:

python --> /usr/bin/python3.6
python3 --> /usr/bin/python3.6
python3.7 --> /usr/bin/python3.7

I install the statsmodels dependencies using `pip3.7`, which means that when I start the interpreter with python3.7 they are all available.  When I use the `pytest` command it uses the correct python interpreter as well (/usr/bin/python3.7).
However, when I try and run the statsmodels test suite I get the two following errors:

python --> /usr/bin/python3.6

```
____________________________________________________________________________________________________________________________ test_lazy_imports ____________________________________________________________________________________________________________________________

    def test_lazy_imports():
        # Check that when statsmodels.api is imported, matplotlib is _not_ imported
        cmd = ("import statsmodels.api as sm; "
               "import sys; "
               "mods = [x for x in sys.modules if 'matplotlib.pyplot' in x]; "
               "assert not mods, mods")
    
        # TODO: is there a cleaner way to do this import in an isolated environment
        pyexe = 'python3' if not PLATFORM_WIN else 'python'
        p = subprocess.Popen(pyexe + ' -c "' + cmd + '"',
                             shell=True, close_fds=True)
        p.wait()
        rc = p.returncode
>       assert rc == 0
E       assert 1 == 0

statsmodels/tests/test_package.py:22: AssertionError
-------------------------------------------------------------------------------------------------------------------------- Captured stderr call ---------------------------------------------------------------------------------------------------------------------------
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/statsmodels/statsmodels/api.py", line 5, in <module>
    from . import iolib
  File "/home/statsmodels/statsmodels/iolib/__init__.py", line 1, in <module>
    from .foreign import StataReader, genfromdta, savetxt
  File "/home/statsmodels/statsmodels/iolib/foreign.py", line 14, in <module>
    from statsmodels.compat.python import (lzip, lmap, lrange,
  File "/home/statsmodels/statsmodels/compat/__init__.py", line 1, in <module>
    from statsmodels.tools._testing import PytestTester
  File "/home/statsmodels/statsmodels/tools/__init__.py", line 1, in <module>
    from .tools import add_constant, categorical
  File "/home/statsmodels/statsmodels/tools/tools.py", line 4, in <module>
    import numpy as np
ModuleNotFoundError: No module named 'numpy'
___________________________________________________________________________________________________________________ test_docstring_optimization_compat ____________________________________________________________________________________________________________________

    @pytest.mark.skipif(SCIPY_11, reason='SciPy raises on -OO')
    def test_docstring_optimization_compat():
        # GH#5235 check that importing with stripped docstrings doesn't raise
        pyexe = 'python3' if not PLATFORM_WIN else 'python'
        p = subprocess.Popen(pyexe + ' -OO -c "import statsmodels.api as sm"',
                             shell=True,
                             stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        out = p.communicate()
        rc = p.returncode
>       assert rc == 0, out
E       AssertionError: (b'', b'Traceback (most recent call last):
E           File "<string>", line 1, in <module>
E           File "/home/statsmodels/statsmod...models/tools/tools.py", line 4, in <module>
E             import numpy as np
E         ModuleNotFoundError: No module named \'numpy\'
E         ')
E       assert 1 == 0

statsmodels/tests/test_package.py:34: AssertionError
```

The errors come about because the `subprocess.Popen` command doesn't use the same interpreter as the parent that's calling it. Here it's resolving to `/usr/bin/python3.6` where none of the dependencies are installed; it should be using `/usr/bin/python3.7`. Not sure what the best way of finding out the current interpreter location is, possibly `sys.executable`?

--- a/statsmodels/tests/test_package.py
+++ b/statsmodels/tests/test_package.py
@@ -1,8 +1,8 @@
 import subprocess
+import sys
 
 import pytest
 
-from statsmodels.compat.platform import PLATFORM_WIN
 from statsmodels.compat.scipy import SCIPY_11
 
 
@@ -12,11 +12,8 @@ def test_lazy_imports():
            "import sys; "
            "mods = [x for x in sys.modules if 'matplotlib.pyplot' in x]; "
            "assert not mods, mods")
-
-    # TODO: is there a cleaner way to do this import in an isolated environment
-    pyexe = 'python3' if not PLATFORM_WIN else 'python'
-    p = subprocess.Popen(pyexe + ' -c "' + cmd + '"',
-                         shell=True, close_fds=True)
+    cmd = sys.executable + ' -c "' + cmd + '"'
+    p = subprocess.Popen(cmd, shell=True, close_fds=True)
     p.wait()
     rc = p.returncode
     assert rc == 0
@@ -25,10 +22,9 @@ def test_lazy_imports():
 @pytest.mark.skipif(SCIPY_11, reason='SciPy raises on -OO')
 def test_docstring_optimization_compat():
     # GH#5235 check that importing with stripped docstrings doesn't raise
-    pyexe = 'python3' if not PLATFORM_WIN else 'python'
-    p = subprocess.Popen(pyexe + ' -OO -c "import statsmodels.api as sm"',
-                         shell=True,
-                         stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    cmd = sys.executable + ' -OO -c "import statsmodels.api as sm"'
+    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE,
+                         stderr=subprocess.PIPE)
     out = p.communicate()
     rc = p.returncode
     assert rc == 0, out

